test/pthread/main.c

```diff
@@ -84,6 +84,67 @@ static int test_mutex_with_concurrent_counter(void) {
     return 0;
 }
 
+// ============================================================================
+// The test case of robust mutex
+// ============================================================================
+
+struct thread_robust_arg {
+    int                         ti;
+    volatile int                *global_count;
+    pthread_mutex_t             *mutex;
+};
+
+static void *thread_worker(void *_arg) {
+    struct thread_robust_arg *arg = _arg;
+    int err = pthread_mutex_lock(arg->mutex);
+    if (err == EOWNERDEAD) {
+        // The next owner should call pthread_mutex_consistent to make mutex consistent again
+        pthread_mutex_consistent(arg->mutex);
```

Comment:
I think after `pthread_mutex_consistent` return, we should try `pthread_mutex_lock`  again.

Reply:
No, this is the common usage, when it calls pthread_mutex_consistent returns, then the thread hold the lock.